pkg/authorization/admission/restrictusers/subjectchecker.go

```diff
@@ -0,0 +1,247 @@
+package restrictusers
+
+import (
+	"fmt"
+
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/labels"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	oclient "github.com/openshift/origin/pkg/client"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// SubjectChecker determines whether rolebindings on a subject (user, group, or
+// service account) are allowed in a namespace.
+type SubjectChecker interface {
```

Comment:
There should be a struct for each matcher type that fulfills this interface.  I'm ok also passing through a context (struct like you have below), but I still can't reasonably read through what this is trying to do with its information.